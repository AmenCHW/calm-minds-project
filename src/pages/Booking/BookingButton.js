import React from 'react';
import { Link } from 'react-router-dom';



  function BookingButton() {

        return (
             <div className="Booking">
              <Link to="/bookingPage" className="focus:text-[#FEE89E]">
                <button  
                  className="mt-20 bg-blue-500 hover:bg-blue-400 text-white 
                  font-bold py-2 px-4 border-b-4 border-blue-700 hover:border-blue-500 rounded" 
                  type="button" >
                   BOOK AN APPOINTMENT    
                </button> 
              </Link>
             
            </div>
        );
}

export default BookingButton;