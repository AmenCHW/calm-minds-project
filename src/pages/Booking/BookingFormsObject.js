export const counselingTypeObject1 = [
  {
    id: 1,
    value: 'Individual',
    label: 'Individual Counseling',
  },
  {
    id: 2,
    value: 'Teen',
    label: 'Teen Counseling (For My Child)',
  },
];

export const relationshipStatusObject2 = [
  {
    id: 3,
    value: 'Single',
    label: 'Single',
  },
  {
    id: 4,
    value: 'Married',
    label: 'Married',
  },
  {
    id: 5,
    value: 'Divorced',
    label: 'Divorced',
  },
];

export const therapyBeforeObject3 = [
  {
    id: 6,
    value: 'Yes',
    label: 'Yes',
  },
  {
    id: 7,
    value: 'No',
    label: 'No',
  },
];

export const SpecificQualitiesObject4 = [
  {
    id: 1,
    text: 'I Prefer A Male Counselor',
    value: 'Male Counselor',
  },
  {
    id: 2,
    text: 'I Prefer A Female Counselor',
    value: 'Female Counselor',
  },
  {
    id: 3,
    text: 'I Prefer An Older Counselor (45+)',
    value: 'Counselor older than 45',
  },
  {
    id: 4,
    text: 'I Prefer A Non-Religious Counselor',
    value: 'Non-Religious Counselor',
  },
  {
    id: 5,
    text: 'I Prefer An Older Counselor',
    value: 'Older Counselor',
  },
];

export const issuesObject5 = [
  {
    id: 6,
    text: 'Depression',
    value: 'Depression',
  },
  {
    id: 7,
    text: 'Stress And Anxiety',
    value: 'Stress And Anxiety',
  },
  {
    id: 8,
    text: 'Relationship issues',
    value: 'Relationship issues',
  },
  {
    id: 9,
    text: 'Family Conflicts',
    value: 'Family Conflicts',
  },
  {
    id: 10,
    text: 'Trauma And Abuse',
    value: 'Trauma And Abuse',
  },
  {
    id: 11,
    text: 'Eating Disorders',
    value: 'Eating Disorders',
  },
];
