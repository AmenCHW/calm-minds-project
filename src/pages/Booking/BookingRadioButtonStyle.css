.form-check input[type='radio'] {
  display: none;
}

.form-check input[type='radio']:checked + label {
  background: #2dd3e3;
  color: white;
}

.form-check label,
.form-check input {
  align-items: center;
  display: flex;
  width: auto;
  height: 313%;
  border-radius: 4px;
  border: #fff;
  padding: 4px 24px;
  cursor: pointer;
  color: #343434;
  background-color: #fff;
  box-shadow: 0px 1px 5px rgb(0 0 0 / 20%);
}
