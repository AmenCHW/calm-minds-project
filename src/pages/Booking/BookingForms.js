import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { counselingTypeObject1, relationshipStatusObject2, therapyBeforeObject3, SpecificQualitiesObject4, issuesObject5 } from './BookingFormsObject';


function BookingForms({ step, setStep, formData, setFormData, handleSubmit }) {
    const [checked, setChecked] = useState('')

    const ChangeFormQuestion = () => {
        let formQuestion;

        if (step === 1) {
            formQuestion = "What Type Of Counseling Are You Looking For?";
        } else if (step === 2) {
            formQuestion = "What Is Your Relationship Status?";
        } else if (step === 3) {
            formQuestion = "Have You Ever Been In Therapy Before?";
        } else if (step === 4) {
            formQuestion = "Are There Any Specific Qualities That You'd Like In A Counselor?";
        } else if (step === 5) {
            formQuestion = "Are There Any Issues You'd Like To Focus On?";
        } else if (step === 7) {
            formQuestion = "Submit Appointment?";
        } else if (step === 8) {
            formQuestion = "Request Submitted";
        }
        // eslint-disable-next-line
        // console.log(formQuestion)
        return formQuestion;
    }

    // eslint-disable-next-line
    // console.log("step", step)

    // eslint-disable-next-line
    // console.log("formData", formData)

    function FormStyle() {
        let formstyle;
        if (step < 6 || step > 6) {
            formstyle = 'p-6 mt-9 mb-12 mx-auto w-auto md:w-[700px] h-auto sm:h-[500px] border-2 border-[#E5E5E5] rounded-md shadow-lg bg-white'
        }
        return formstyle;
    }

    function FormButtton() {
        let button;
        if (step <= 6) {
            button = <button type="submit"
                disabled={step === 1 && formData.counseling_type === null || step === 2 && formData.relationship_status === null || step === 3 && formData.ever_been_in_therapy_before === null || step === 4 && formData.counselor_qualities === null || step === 5 && formData.issues === null || step === 6 && (formData.what_brings_you_here === null || !formData.what_brings_you_here.trim().length)}
                onClick={() => { setStep((currentStep) => currentStep + 1) }
                }
                className='bg-[#2DD3E3] text-1xl sm:text-2xl px-6 py-3 font-normal border-2 border-[#2DD3E3] rounded-md place-self-start mt-5' >
                NEXT
            </button >
        } else if (step === 7) {
            button = <button type="submit"
                onClick={() => { setStep((currentStep) => currentStep + 1) }}
                className='bg-[#2DD3E3] text-1xl sm:text-2xl px-12 sm:px-28 py-3 font-normal border-2 border-[#2DD3E3] place-self-center items-center justify-center mx-auto rounded-md mt-7 mb-1 sm:mb-0'>
                SUBMIT
            </button>
        } else if (step === 8) {
            button = <button type="submit"
                className='bg-[#2DD3E3] text-1xl sm:text-2xl px-8 sm:px-16 py-3 font-normal border-2 border-[#2DD3E3] rounded-md place-self-center items-center justify-center self-center mx-auto mt-7 mb-1 sm:mb-0'>
                <Link to="/"> BACK TO HOME </Link>
            </button>
        }
        return button;
    }

    useEffect(() => {
        FormStyle();
        FormButtton();
    }, [step]);

    function DisplayRadioButtons() {
        let radioButtons;
        if (step === 1) {
            radioButtons = counselingTypeObject1.map(({ id, value, label }) => {
                return (
                    <div className="form-check pt-4 text-2xl sm:text-3xl font-normal" key={id}>
                        <input
                            type="radio"
                            name="exampleRadios"
                            id={id}
                            value={value}
                            checked={value === checked}
                            onChange={() => { setChecked(value); setFormData({ ...formData, counseling_type: value }) }}
                        />
                        <label className="form-check-label" htmlFor={id}>
                            {label}
                        </label>
                    </div>
                )
            })
        } else if (step === 2) {
            radioButtons = relationshipStatusObject2.map(({ id, value, label }) => {
                return (
                    <div className="form-check pt-4 text-2xl sm:text-3xl font-normal" key={id}>
                        <input
                            type="radio"
                            name="exampleRadios"
                            id={id}
                            value={value}
                            checked={value === checked}
                            onChange={() => { setChecked(value); setFormData({ ...formData, relationship_status: value }) }}
                        />
                        <label className="form-check-label" htmlFor={id}>
                            {label}
                        </label>
                    </div>
                )
            })
        } else if (step === 3) {
            radioButtons = therapyBeforeObject3.map(({ id, value, label }) => {
                return (
                    <div className="form-check pt-4 text-2xl sm:text-3xl font-normal" key={id}>
                        <input
                            type="radio"
                            name="exampleRadios"
                            id={id}
                            value={value}
                            checked={value === checked}
                            onChange={() => { setChecked(value); setFormData({ ...formData, ever_been_in_therapy_before: value }) }}
                        />
                        <label className="form-check-label" htmlFor={id}>
                            {label}
                        </label>
                    </div>
                )
            })
        } else if (step === 4) {
            radioButtons = SpecificQualitiesObject4.map(({ id, value, text }) => {
                return (
                    <div key={id} className="flex py-3">
                        <input
                            key={id}
                            type="radio"
                            htmlFor="for"
                            id={id}
                            name="type"
                            value={value}
                            checked={value === checked}
                            onChange={() => { setChecked(value); setFormData({ ...formData, counselor_qualities: value }) }}
                        />
                        <p className="ml-3 text-lg sm:text-xl">{text}</p>
                    </div>
                );
            })
        } else if (step === 5) {
            radioButtons = issuesObject5.map(({ id, value, text }) => {
                return (
                    <div key={id} className="flex py-3">
                        <input
                            key={id}
                            type="radio"
                            htmlFor="for"
                            id={id}
                            name="type"
                            value={value}
                            checked={value === checked}
                            onChange={() => { setChecked(value); setFormData({ ...formData, issues: value }) }}
                        />
                        <p className="ml-3 text-lg sm:text-xl">{text}</p>
                    </div>
                );
            })
        } else if (step === 6) {
            radioButtons = <input
                type="text"
                name="paragraph"
                onChange={(e) => setFormData({ ...formData, what_brings_you_here: e.target.value })}
                className="border-2 border-[#E5E5E5] rounded-md p-2 w-full h-[360px] mx-auto"
                required
            />
        }
        return radioButtons;
    }


    return (

        <div>
            <form onSubmit={handleSubmit} className={FormStyle()}>
                <h1 className={`text-2xl sm:text-3xl font-normal pb-8 ${step > 6 ? "text-center sm:pt-16 sm:pb-10" : "text-left"}`} >
                    {ChangeFormQuestion()}
                </h1>

                {step > 6 && <p className='text-center text-xl sm:text-2xl px-2 sm:px-16 sm:py-10'>{step === 7 ? "Please Be Aware That This Action Will Cost You A Ticket!" : "You Will Receive A Confirmation Email Soon Please Keep An Eye On Your Mail."} </p>}

                <div className='flex flex-col'>
                    {DisplayRadioButtons()}
                    {FormButtton()}
                </div>

            </form>
        </div >

    )
}

export default BookingForms;

