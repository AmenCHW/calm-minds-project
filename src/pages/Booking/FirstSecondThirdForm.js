import React, { useState } from 'react';
import "./BookingRadioButtonStyle.css"




function FirstSecondThirdForm({ step, setStep, formData, setFormData }) {
    const [checked, setChecked] = useState('')

    const radioButtons1 = [{
        id: 1,
        value: 'Individual',
        label: 'Individual Counseling',
        title: 'counseling_type',
    },
    {
        id: 2,
        value: 'Teen',
        label: 'Teen Counseling (For My Child)',
        title: 'counseling_type',
    },
    ]

    const radioButtons2 = [{
        id: 3,
        value: 'Single',
        label: 'Single',
        title: 'relationship_status',
    },
    {
        id: 4,
        value: 'Married',
        label: 'Married',
        title: 'relationship_status',
    },
    {
        id: 5,
        value: 'Divorced',
        label: 'Divorced',
        title: 'relationship_status',

    },
    ]

    const radioButtons3 = [{
        id: 6,
        value: 'Yes',
        label: 'Yes',
        title: 'ever_been_in_therapy_before'
    },
    {
        id: 7,
        value: 'No',
        label: 'No',
        title: 'ever_been_in_therapy_before'
    },
    ]

    const handleSubmit = (e) => {
        e.preventDefault();
    }


    const ChangeFormQuestion = () => {
        let formQuestion;

        if (step === 1) {
            formQuestion = "What Type Of Counseling Are You Looking For?";
        } else if (step === 2) {
            formQuestion = "What Is Your Relationship Status?";
        } else if (step === 3) {
            formQuestion = "Have You Ever Been In Therapy Before?";
        }
        // } else if (step === 4) {
        //   formQuestion = "Are There Any Specific Qualities That You'd Like In A Counselor?";
        // } else if (step === 5) {
        //   formQuestion = "Are There Any Issues You'd Like To Focus On?";
        // }
        // eslint-disable-next-line
        console.log(formQuestion)
        return formQuestion;
    }

    // eslint-disable-next-line
    console.log("step", step)

    function DisplayRadioButtons() {
        let radioButtons;
        if (step === 1) {
            radioButtons = radioButtons1
        } else if (step === 2) {
            radioButtons = radioButtons2
        } else if (step === 3) {
            radioButtons = radioButtons3
        }
        return radioButtons.map(({ id, value, label, title }) => {
            // eslint-disable-next-line
            console.log(title)
            return (
                <div className="form-check pt-4 text-1xl sm:text-3xl font-normal" key={id}>
                    <input
                        type="radio"
                        name="exampleRadios"
                        id={id}
                        value={value}
                        checked={value === checked}
                        onChange={() => { setChecked(value); setFormData({ ...formData, title: value }) }}
                    />
                    <label className="form-check-label" htmlFor={id}>
                        {label}
                    </label>
                </div>
            )
        });
    }

    return (

        <div className='p-6 mt-9 mb-12 mx-auto w-auto md:w-[700px]  h-[500px] border-2 border-[#E5E5E5] rounded-md shadow-lg bg-white'>
            <form onSubmit={handleSubmit}>
                <h1 className='text-1xl sm:text-3xl font-normal pb-8 text-center md:text-left'>
                    {ChangeFormQuestion()}
                </h1>

                <div className='flex flex-col'>
                    {DisplayRadioButtons()}

                    <button type="submit"
                        disabled={step > 3}
                        onClick={() => { setStep((currentStep) => currentStep + 1) }}
                        className='bg-[#2DD3E3] text-1xl sm:text-2xl px-6 py-3 font-normal border-2 border-[#2DD3E3] rounded-md place-self-start relative left-0 top-[12em] sm:top-[7em] md:top-[8em] lg:top-[8.5em]'>
                        NEXT
                    </button>
                </div>
            </form>
        </div>

    )
}

export default FirstSecondThirdForm;



