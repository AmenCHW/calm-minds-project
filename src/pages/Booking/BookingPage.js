import React, { useState, useEffect } from 'react';
import { doc, updateDoc } from "firebase/firestore";
import { db } from '../../firebase-config';
import { UserAuth } from '../../context/AuthContext';
import BookingForms from './BookingForms';
import "./BookingRadioButtonStyle.css"

function Booking() {

  const { user } = UserAuth();

  // eslint-disable-next-line
  { user && console.log('user sent', user) }

  const [step, setStep] = useState(1)
  const [pageTitle, setPageTitle] = useState('')
  const [pageText, setPageText] = useState('')

  const [formData, setFormData] = useState({
    counseling_type: null,
    relationship_status: null,
    ever_been_in_therapy_before: null,
    counselor_qualities: null,
    issues: null,
    what_brings_you_here: null,
  })

  const handleSubmit = (e) => {
    e.preventDefault();
  }


  useEffect(() => {
    if (step < 6) {
      setPageTitle("LET'S MATCH YOU WITH THE RIGHT THERAPIST");
      setPageText("Please fill out this short questionnaire to provide some general and anonymous background about you and the issues you'd like to deal with in online therapy. It would help us match you with the most suitable therapist for you.");
    } else if (step === 6) {
      setPageTitle("WHAT BRINGS YOU HERE?");
      setPageText("Please specify (in a few sentences) why you'd like counseling. This will give your counselor a good understanding of where to start.")
    } else if (step === 7) {
      setPageTitle("SUBMIT YOUR APPOINTMENT");
      setPageText("Click Submit if you are sure of all your choices.");
    } else if (step === 8) {
      setPageTitle("YOUR REQUEST HAS BEEN SUBMITTED");
      setPageText("You will receive an email guiding you to book a date and time soon.");
    }
  }, [step]);

  useEffect(() => {
    if (!user) {
      // eslint-disable-next-line
      alert("please signup before booking an appointment")
    }
    if (user && user.uid)
      updateDoc(doc(db, 'users', user.uid), {
        book_an_appointment: {
          counseling_type: formData.counseling_type,
          relationship_status: formData.relationship_status,
          ever_been_in_therapy_before: formData.ever_been_in_therapy_before,
          counselor_qualities: formData.counselor_qualities,
          issues: formData.issues,
          what_brings_you_here: formData.what_brings_you_here,
        }
      }).then(response => {
        // eslint-disable-next-line
        console.log(response)
        // eslint-disable-next-line
        console.log("updated")
      }).catch(error => {
        // eslint-disable-next-line
        console.log(error.message)
      })
  }, [formData])



  return (
    <div className='mx-auto lg:max-w-7xl px-10 py-10'>

      <h1 className='pt-5 text-3xl md:text-5xl format-normal leading-normal'>
        {pageTitle}
      </h1>

      <p className='mt-2 text-2xl text-gray-500'>
        {pageText}
      </p>

      <div>
        <BookingForms step={step} setStep={setStep} formData={formData} setFormData={setFormData} handleSubmit={handleSubmit} />
      </div>

    </div>
  );
}

export default Booking;
