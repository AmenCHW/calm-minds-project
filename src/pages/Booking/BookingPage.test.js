import React from "react";
 // import {render,screen } from "@testing-library/react"
import Booking from "./BookingPage";
import renderer from 'react-test-renderer' 

test("first snapshot testing",()=>{
    const component=renderer.create( <Booking/>).toJSON()  
    expect( component).toMatchSnapshot();
})





