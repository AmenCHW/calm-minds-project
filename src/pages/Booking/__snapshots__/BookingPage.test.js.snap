// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot testing 1`] = `
<div
  className="mx-auto lg:max-w-7xl px-10 py-10"
>
  <h1
    className="pt-5 text-3xl md:text-5xl format-normal leading-normal"
  />
  <p
    className="mt-2 text-2xl text-gray-500"
  />
  <div>
    <div>
      <form
        className="p-6 mt-9 mb-12 mx-auto w-auto md:w-[700px] h-auto sm:h-[500px] border-2 border-[#E5E5E5] rounded-md shadow-lg bg-white"
        onSubmit={[Function]}
      >
        <h1
          className="text-2xl sm:text-3xl font-normal pb-8 text-left"
        >
          What Type Of Counseling Are You Looking For?
        </h1>
        <div
          className="flex flex-col"
        >
          <div
            className="form-check pt-4 text-2xl sm:text-3xl font-normal"
          >
            <input
              checked={false}
              id={1}
              name="exampleRadios"
              onChange={[Function]}
              type="radio"
              value="Individual"
            />
            <label
              className="form-check-label"
              htmlFor={1}
            >
              Individual Counseling
            </label>
          </div>
          <div
            className="form-check pt-4 text-2xl sm:text-3xl font-normal"
          >
            <input
              checked={false}
              id={2}
              name="exampleRadios"
              onChange={[Function]}
              type="radio"
              value="Teen"
            />
            <label
              className="form-check-label"
              htmlFor={2}
            >
              Teen Counseling (For My Child)
            </label>
          </div>
          <button
            className="bg-[#2DD3E3] text-1xl sm:text-2xl px-6 py-3 font-normal border-2 border-[#2DD3E3] rounded-md place-self-start mt-5"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
            NEXT
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
